### DISCUSSION TITLE
data: protocol in img tag src attribute not allowed for basicWithImages by default

### DISCUSSION THREAD

**Discussion by RaphiD** [jhy/jsoup#2132] [Q&A]
Hi, why is the <code>data</code> protocol not allowed by default if I use the <code>basicWithImages</code> safe list to clean HTML content?

Javadoc:
This safelist allows the same text tags as {@link #basic}, and also allows <code>img</code> tags, with appropriate attributes, with <code>src</code> pointing to <code>http</code> or <code>https</code>.

**Comment by ecki** [comment]
Specifically, we wonder if there is an additional risk involved (like the <img src=data:> can contain js and is miss-interpreted by (older) browsers or src=data: might be somehow re-used for <script>. Asuming JSoup gets that right, is it safe to allow it - or would you conider adding it to basic protocols? (I mean data: sounds less dangerous as any untrusted remote url, right?)

**Comment by MasterChiefNemo** [comment]
Would I be correct in saying the safelist is used to sanitize data being parsed by JSoup?

**Reply by ecki** [reply]
What do you mean? The safelist which allows image tags does not allow them if they are inline. The question is why

**Comment by jhy** [comment]
I think it would be a fine thing to add support for. We would need to review and see what kind (if any) validation needs to occur on the data (like e.g. checking they are all in the expected character range, etc).

I would like to add a new set of default safelists that we can extend over time. I am reluctant to make changes to the existing default safelists. They can be named like HTML5, based on the concept that HTML5 itself is evolving.

See also #1297 which is related.
